compendium/DeclarativeServices/test/TestSingletonComponentConfiguration.cpp

```diff
@@ -205,6 +227,51 @@ TEST_F(SingletonComponentConfigurationTest, TestGetService)
     instCtxtPair->second.reset();
   }
 }
+/* This test verifies that if the Modified method of a component instance throws an 
+ * exception DS intercepts the exception and logs it. 
+ */
+TEST_F(SingletonComponentConfigurationTest,
+       TestModifiedMethodExceptionLogging)
+{
+  using cppmicroservices::logservice::SeverityLevel;
+
+  auto mockCompContext = std::make_shared<MockComponentContextImpl>(obj);
+  auto mockCompInstance = std::make_shared<MockComponentInstance>();
+  obj->SetComponentInstancePair(InstanceContextPair(
+    mockCompInstance,
+    mockCompContext)); 
+  EXPECT_NE(obj->GetComponentInstance(), nullptr);
+  EXPECT_NE(obj->GetComponentContext(), nullptr);
+
+  // set logging expectations
+  auto ExceptionThrownByModifiedMethod =
+    testing::AllOf(
+      testing::HasSubstr("Exception received from user code while modifying "),
+      testing::HasSubstr("component configuration"));
+  EXPECT_CALL(*mockLogger,
+              Log(SeverityLevel::LOG_ERROR, ExceptionThrownByModifiedMethod, testing::_))
+    .Times(1);
+
+  // When the mock Modified method is called it will throw the ModifiedMethodException
+  EXPECT_CALL(*mockCompInstance,Modified()).Times(1).WillRepeatedly(ModifiedMethodException());
+
+  // When the mock DoesModifiedMethodExist method is called it will return true;
+  EXPECT_CALL(*mockCompInstance, DoesModifiedMethodExist())
+    .Times(1)
+    .WillRepeatedly(ModifiedMethodExists());
+
+  // Deactivate and Unbindreference will also be called. EXPECT_CALL added
+  // to avoid GMOCK Warning. 
+   // When the mock Modified method is called it will throw the ModifiedMethodException
+  EXPECT_CALL(*mockCompInstance, Deactivate()).Times(1);
+  EXPECT_CALL(*mockCompInstance, UnbindReferences()).Times(1);
```

Comment:
GMOCK_WARNING:
Uninteresting mock function call - returning directly. Function call Deactivate()
Note: You can safely ignore the above warning unless this call should not happen. Do not suppress it by blindly adding an EXPECT_CALL() if  you don't mean to enforce the call. See https://github.com/google/googletest/blob/master/googlemock/docs/cook_book.md#knowing-when-to-expect for details. 

The second warning looks just like that except the Function call is UnbindReferences. 

The warnings are coming up because Deactivate and UnbindReferences were called but we didn't have any EXPECT_CALL statements to indicate that we expected those to be called. It is valid that they are called during this test so I added the EXPECT_CALL statements to indicate that.

Reply:
Ok. Unless you want to test that Deactivate adn UnbindReferences are called in this testpoint, adding the EXPECT_CALL is unnecessary since, as the warning says: `You can safely ignore the above warning unless this call should not happen. Do not suppress it by blindly adding an EXPECT_CALL() if you don't mean to enforce the call.`. 

I'm fine with leaving the `EXPECT_CALL`s in however this warning is meant to flag the user about possible unenforced expectations on the mock. We have this warning in other testpoints. 

Another solution to get rid of the warning is outlined here: https://google.github.io/googletest/gmock_cook_book.html#NiceStrictNaggy.